mautic/mautic
# Add groupBy to contacts list

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for 4.0
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 




#### Description:
This is very rare issue and it's related to Mysql 5.7.33
We had 86 contacts with null l.last_active. 
We had prefer_ordering_index=on in optimizer_switch of Mysql


https://dev.mysql.com/doc/refman/5.7/en/switchable-optimizations.html#optflag_prefer-ordering-index

This option try optimize order by If cannot find values.

But different ordered by results for 

- `SELECT l.* FROM leads l LEFT JOIN users u ON u.id = l.owner_id WHERE l.date_identified IS NOT NULL ORDER BY l.last_active DESC LIMIT 30 OFFSET 0`

vs
 
- `SELECT l.* FROM leads l LEFT JOIN users u ON u.id = l.owner_id WHERE l.date_identified IS NOT NULL ORDER BY l.last_active DESC LIMIT 50 OFFSET 0`




#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. It's hard to reproduce
3. Just code review , this should not affect anybody




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10122"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
